### PR TITLE
sh: make cluster version configuration unified

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
sh: make cluster version configuration unified

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
LGTM

**Comment by chaosi-zju** [comment]
when can we use v1.28 ?

**Comment by jwcesign** [comment]
> when can we use v1.28 ?

I am not sure.  Let's first consolidate all version modifications.

File: hack/util.sh [hunk 1]
```diff
@@ -25,6 +25,8 @@ KARMADA_GO_PACKAGE="github.com/karmada-io/karmada"
 
 MIN_Go_VERSION=go1.20.0
 
+DEFAULT_CLUSTER_VERSION="kindest/node:v1.27.3"
```
Comment (RainbowMango) [inline hack/util.sh:28]: It makes sense to take this as the single source of truth, so when next time we bump kind, we only need to make an update here. Is it the idea?

Comment (jwcesign) [reply hack/util.sh:28]: Yes, that's right.

**Review by RainbowMango — APPROVED** [review]
/approve

**Comment by chaunceyjiang** [comment]
image

My MAC m1 is not working. @jwcesign

**Pull request merged**